### DISCUSSION TITLE
[Q] how can i format num by numFmt when i use exceljs to read xlsx (!is read not write)

### DISCUSSION THREAD

**Discussion by 3kinglee** [exceljs/exceljs#2428] [Q&A]
when i user exceljs to read a .xlsx file, i can get a cell value and cell numFmt, but how can i user this to format number。

can you provide an api like this:
```
// ( '25',  '0.00%' ) => 250.00%

const numString = numFormat(numValue, numFmt)

```

**Comment by ketakale321** [comment]
Also when I try to set the 'style' property like

worksheet.getRow(i).getCell(j).style={numFmt:'@'};

and then try to access the cell value like this:

let value = worksheet.getRow(i).getCell(j).value;

then I get the cell value in Number formant only.

Please refer https://github.com/exceljs/exceljs/issues/528 for numFmt:'@'

**Comment by ketakale321** [comment]
Hi All,
I found one temporary fi for this.
While reading the file itself I used map option.

From docs i found and modified it to like this:

// read from a file with custom value parsing
const workbook = new Excel.Workbook();
const options = {
  map(value, index) {
    switch(index) {
      default:
        return JSON.stringify(value);
    }
  },
const worksheet = await workbook.csv.readFile(filename, options);

It was returning all the columns to me in text format.

**Comment by ShraddhaJ11** [comment]
How to retrieve the Cell Text as Displayed in Excel?
for currency 340 , displayed value is $340.00 but cell.value and cell.text both return 340 only. So how to get the value with number format?

**Comment by ShraddhaJ11** [comment]
> when i user exceljs to read a .xlsx file, i can get a cell value and cell numFmt, but how can i user this to format number。
> 
> can you provide an api like this:
> 
> ```
> // ( '25',  '0.00%' ) => 250.00%
> 
> const numString = numFormat(numValue, numFmt)
> ```
@three-king Did you find any solution for this?

**Comment by yoya93** [comment]
> > cuando utilizo exceljs para leer un archivo .xlsx, puedo obtener un valor de celda y un numFmt de celda, pero ¿cómo puedo usar esto para formatear el número?
> > ¿Puedes proporcionar una API como esta?
> > ```
> > // ( '25',  '0.00%' ) => 250.00%
> > 
> > const numString = numFormat(numValue, numFmt)
> > ```
> 
> @ three-king ¿Encontraste alguna solución para esto?

@ShraddhaJ11 Did you find any solution for this?

**Comment by OscarDTorresB** [comment]
Is this still a thing? I am looking for something like this

**Comment by protobi-pieter** [comment]
@OscarDTorresB @yoya93 @3kinglee 

There is a library that does exactly this, https://github.com/borgar/numfmt, interpreting Excel number formats( OOXML ECMA-376), e.g.

`let  formattedText = numfmt.format(cell.numFmt, cell.value) // numfmt.format( '#,##0.0', 2391.22348) => 2391.2`
